crypto/signature.go

```diff
@@ -25,6 +25,19 @@ func VerifySignedMessageByOwner(
 	}
 
 	if isEOASignature {
+		// EIP 155 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+		if signature[64] == 27 || signature[64] == 28 {
+			//{0,1} + 27
+			signature[64] -= byte(27)
+		} else if signature[64] != 0 && signature[64] != 1 {
+			//{0,1} + CHAIN_ID * 2 + 35
+			chainID, err := client.NetworkID(context.Background())
+			if err != nil {
+				return err
+			}
+			signature[64] -= byte(chainID.Uint64()*2 + 35)
+		}
+
 		pk, err := eth_crypto.SigToPub(hash[:], signature)
```

Comment:
So eth_crypto.SigToPub also adapts to this change and it recovers the correct pk with EIP 155 signatues?